infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/fsm/NegotiatingState.java

```diff
@@ -73,7 +77,13 @@ public LogReplicationRuntimeState processEvent(LogReplicationRuntimeEvent event)
                 return null;
             case NEGOTIATION_COMPLETE:
                 log.info("Negotiation complete, result={}", event.getNegotiationResult());
-                ((ReplicatingState)fsm.getStates().get(LogReplicationRuntimeStateType.REPLICATING)).setReplicationEvent(event.getNegotiationResult());
+                if(tableManagerPlugin.isUpgraded()) {
+                    // Force a snapshot sync if upgrading is detected
```

Comment:
We can improve this message, for instance:

Force a snapshot sync if an upgrade has been identified. This will guarantee that changes in the streams to replicate are captured by the destination.

Reply:
Done, thanks!